src/components/views/rooms/PinnedEventsPanel.js

```diff
@@ -71,6 +71,25 @@ module.exports = React.createClass({
                 this.setState({ loading: false, pinned });
             });
         }
+
+        this._updateReadState();
+    },
+
+    _updateReadState: function() {
+        const pinnedEvents = this.props.room.currentState.getStateEvents("m.room.pinned_events", "");
+        if (!pinnedEvents) return; // nothing to read
+
+        let lastReadEvent = null;
+        const readPinsEvent = this.props.room.getAccountData("im.vector.room.read_pins");
+        if (readPinsEvent) {
+            lastReadEvent = readPinsEvent.getContent().last_read_id;
+        }
+
+        if (lastReadEvent !== pinnedEvents.getId()) {
```

Comment:
This doesn't make sure that the `lastReadEvent` is ahead in the DAG relative to `pinnedEvents.getId()`. This could lead to two clients (i.e. clients with potentially differing state event data) fighting over which pinned message is really the last one read.

Reply:
What's the best way to check for this? Just using `.getTs()`?

Reply:
We can't technically rely on the timestamp to compare events because it might not be accurate. If we want to do this with some guarantee of the event ID always "increasing" in the timeline, we'd need server-side support. This would look something like `m.fully_read` - the read marker.

As there's no generic API for setting a similar `m.pinned_messages.fully_read` or whatever, this would require server-side changes. 

But if we assume that the TS on the events are accurate and comparable (which they probably are), then sure.

Reply:
`m.fully_read` doesn't seem to appear in the spec at all. Is it something that already exists or is that part of the server-side implementation request?

Reply:
Ah.. then you probably want this https://docs.google.com/document/d/1UWqdS-e1sdwkLDUY0wA4gZyIkRp-ekjsLZ8k6g_Zvso/edit

Reply:
(which is https://github.com/matrix-org/matrix-doc/issues/910) 

I'll take a look and give it a shot I suppose.